app/src/test/java/org/oppia/android/app/player/state/StateFragmentLocalTest.kt

```diff
@@ -267,6 +268,67 @@ class StateFragmentLocalTest {
     }
   }
 
+  @Test
+  fun testContinueInteractionAnim_openPrototypeExp_checkContinueButtonAnimatesAfter45Seconds() {
+    TestPlatformParameterModule.forceEnableContinueButtonAnimation(true)
+    launchForExploration(TEST_EXPLORATION_ID_2).use {
+      startPlayingExploration()
+      testCoroutineDispatchers.runCurrent()
+
+      onView(withId(R.id.state_recycler_view)).perform(scrollToViewType(CONTINUE_INTERACTION))
+      onView(withId(R.id.continue_button)).check(matches(Matchers.not(isAnimating())))
+      testCoroutineDispatchers.advanceTimeBy(45000)
+      onView(withId(R.id.continue_button)).check(matches(isAnimating()))
+    }
+  }
+
+  @Test
+  fun testConIntAnim_openProtExp_orientLandscapeAfter30Sec_checkAnimStartsIn45SecAfterOpening() {
+    TestPlatformParameterModule.forceEnableContinueButtonAnimation(true)
+    launchForExploration(TEST_EXPLORATION_ID_2).use {
+      startPlayingExploration()
+      testCoroutineDispatchers.runCurrent()
+
+      scrollToViewType(CONTINUE_INTERACTION)
+      onView(withId(R.id.continue_button)).check(matches(Matchers.not(isAnimating())))
+      testCoroutineDispatchers.advanceTimeBy(30000)
+      onView(isRoot()).perform(orientationLandscape())
+      testCoroutineDispatchers.runCurrent()
+      scrollToViewType(CONTINUE_INTERACTION)
+      onView(withId(R.id.continue_button)).check(matches(Matchers.not(isAnimating())))
+      testCoroutineDispatchers.advanceTimeBy(15000)
+      onView(withId(R.id.continue_button)).check(matches(isAnimating()))
+    }
+  }
+
+  @Test
+  fun testConIntAnim_openFractions_expId1_checkButtonDoesNotAnimate() {
+    TestPlatformParameterModule.forceEnableContinueButtonAnimation(true)
+    launchForExploration(TEST_EXPLORATION_ID_2).use {
+      startPlayingExploration()
+      playThroughTestState1()
+      submitFractionAnswer(answerText = "1/2")
+
+      scrollToViewType(CONTINUE_NAVIGATION_BUTTON)
+      testCoroutineDispatchers.advanceTimeBy(45000)
+      onView(withId(R.id.continue_navigation_button)).check(matches(Matchers.not(isAnimating())))
+    }
+  }
+
+  private fun isAnimating(): TypeSafeMatcher<View> {
+    return ActiveAnimationMatcher()
+  }
+
+  private class ActiveAnimationMatcher() : TypeSafeMatcher<View>() {
+    override fun describeTo(description: Description) {
+      description.appendText("View is animating")
+    }
+
+    override fun matchesSafely(view: View): Boolean {
+      return view.animation?.hasStarted() ?: false
+    }
+  }
```

Comment:
Please move these to be below the other tests where other helpers go.

Reply:
done